backend/PythonClient/server/simulation_server.py

```diff
@@ -33,90 +36,53 @@
-        return 0
+    # Fetch the list of reports from GCS bucket
+    bucket = storage_client.bucket(bucket_name)
+    blobs = bucket.list_blobs(prefix='reports/')
+
+    report_files = []
     monitors = ['CollisionMonitor', 'LandspaceMonitor', 'NoFlyZoneMonitor', 'PointDeviationMonitor']
     global_monitors = ['MinSepDistMonitor']
-    report_files = []
-    for file in os.listdir(reports_path):
-        file_path = os.path.join(reports_path, file)
-        if os.path.isdir(file_path):
-            pass_count = fail_count = 0
-            contains_fuzzy = any('fuzzy' in sub.lower() for sub in os.listdir(file_path))
-            paths_to_check = []
-            if contains_fuzzy:
-                # Check each fuzzy subdirectory
-                for subdir in os.listdir(file_path):
-                    if 'fuzzy' in subdir.lower():
-                        subdir_path = os.path.join(file_path, subdir)
-                        paths_to_check.append(subdir_path)
-            else:
-                # If no fuzzy subdirectory, check the main directory
-                paths_to_check.append(file_path)
-            drone_count = 0
-            for path in paths_to_check:
-                drone_count = max(drone_count, count_drone_files(path))
-            all_monitors_pass = True
-            for path in paths_to_check:
-                flytopoints_path = os.path.join(path, 'FlyToPoints')
-                if os.path.exists(flytopoints_path) and os.path.isdir(flytopoints_path):
-                    for monitor in monitors:
-                        if not check_monitor_pass_fail(flytopoints_path, monitor):
-                            all_monitors_pass = False
-                            break
-                    if not all_monitors_pass:
-                        break
-                global_monitors_path = os.path.join(path, 'GlobalMonitors')
-                if os.path.exists(global_monitors_path) and os.path.isdir(global_monitors_path):
-                    for monitor in global_monitors:
-                        if not check_monitor_pass_fail(global_monitors_path, monitor):
-                            all_monitors_pass = False
-                            break
-                    if not all_monitors_pass:
-                        break
-            if all_monitors_pass:
-                pass_count = drone_count
-            else:
-                fail_count = drone_count
-            report_files.append({
-                'filename': file,
-                'contains_fuzzy': contains_fuzzy,
-                'drone_count': drone_count,
-                'pass': pass_count,
-                'fail': fail_count
-            })
-        else:
+
+    for blob in blobs:
+        if blob.name.endswith('.txt'):
+            file_name = blob.name.split('/')[-1]
+            drone_count = count_drone_files(bucket, file_name)
+            all_monitors_pass = check_monitor_pass_fail(bucket, file_name, monitors)
```

Comment:
The code extracts only the filename (blob.name.split('/')[-1]), losing the full path structure needed for nested directories (like "FlyToPoints").